### ISSUE TITLE
CI tests are flaky due to spurious Maven failures

### ISSUE THREAD

**Issue by dicej** [bytecodealliance/wit-bindgen#495]
CI started failing for various PRs a few days ago, unrelated to any code changes.  I've been unable to reproduce it locally.

Using the latest OpenJDK and latest Maven releases initially seemed to fix it, but that seems to have been a coincidence since it still fails most of the time.

If I can't figure out how to make Maven happy, I'll switch the Java tests to download and run a TeaVM "fat" jar and run it that way, bypassing Maven.

**Comment by dicej** [comment]
For reference, here is an example of the output of a failing test:

```
---- numbers::run stdout ----
wasi adapter = "D:\\a\\wit-bindgen\\wit-bindgen\\target\\debug\\build\\test-artifacts-17eef5ab52e6f25e\\out\\wasm32-unknown-unknown/release/wasi_snapshot_preview1.wasm"
rust core module = "D:\\a\\wit-bindgen\\wit-bindgen\\target\\debug\\build\\test-artifacts-17eef5ab52e6f25e\\out\\wasm32-wasi/debug\\numbers.wasm"
rust component "D:\\a\\wit-bindgen\\wit-bindgen\\target\\runtime-tests\\numbers\\rust.wasm"
"D:/a/wit-bindgen/wit-bindgen/wasi-sdk-16.0\\bin/clang" "--sysroot" "D:/a/wit-bindgen/wit-bindgen/wasi-sdk-16.0\\share/wasi-sysroot" "./tests/runtime\\numbers\\wasm.c" "D:\\a\\wit-bindgen\\wit-bindgen\\target\\runtime-tests\\numbers\\numbers.c" "D:\\a\\wit-bindgen\\wit-bindgen\\target\\runtime-tests\\numbers\\numbers_component_type.o" "-I" "D:\\a\\wit-bindgen\\wit-bindgen\\target\\runtime-tests\\numbers" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-mexec-model=reactor" "-g" "-o" "D:\\a\\wit-bindgen\\wit-bindgen\\target\\runtime-tests\\numbers\\c-wasm.wasm"
"cmd" "/c" "mvn" "prepare-package"
status: exit code: 1
stdout: ------------------------------------------
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------< foo:foo >-------------------------------
[INFO] Building foo 1.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ war ]---------------------------------
Downloading from central: https://repo.maven.apache.org/maven2/com/fermyon/teavm-maven-plugin/0.2.5/teavm-maven-plugin-0.2.5.pom
Progress (1): 1.4/3.8 kB
Progress (1): 2.7/3.8 kB
Progress (1): 3.8 kB    
                    
Downloaded from central: https://repo.maven.apache.org/maven2/com/fermyon/teavm-maven-plugin/0.2.5/teavm-maven-plugin-0.2.5.pom (3.8 kB at 5.2 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/com/fermyon/teavm-maven/0.2.5/teavm-maven-0.2.5.pom
Progress (1): 1.3 kB
                    
Downloaded from central: https://repo.maven.apache.org/maven2/com/fermyon/teavm-maven/0.2.5/teavm-maven-0.2.5.pom (1.3 kB at 42 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/com/fermyon/teavm/0.2.5/teavm-0.2.5.pom
Progress (1): 1.4/16 kB
Progress (1): 2.7/16 kB
Progress (1): 4.1/16 kB
Progress (1): 5.5/16 kB
Progress (1): 6.9/16 kB
Progress (1): 8.2/16 kB
Progress (1): 9.6/16 kB
Progress (1): 11/16 kB 
Progress (1): 12/16 kB
Progress (1): 14/16 kB
Progress (1): 15/16 kB
Progress (1): 16/16 kB
Progress (1): 16 kB   
                   
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.438 s
[INFO] Finished at: 2023-02-10T17:42:53Z
[INFO] ------------------------------------------------------------------------
Error:  Plugin com.fermyon:teavm-maven-plugin:0.2.5 or one of its dependencies could not be resolved: Failed to read artifact descriptor for com.fermyon:teavm-maven-plugin:jar:0.2.5: Could not transfer artifact com.fermyon:teavm:pom:0.2.5 from/to central (https://repo.maven.apache.org/maven2): C:\Users\runneradmin\.m2\repository\com\fermyon\teavm\0.2.5\teavm-0.2.5.pom.16581083516929454297.tmp -> C:\Users\runneradmin\.m2\repository\com\fermyon\teavm\0.2.5\teavm-0.2.5.pom -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

stderr: ------------------------------------------

thread 'numbers::run' panicked at 'failed to build', tests\runtime\main.rs:272:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    numbers::run

test result: FAILED. 8 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 77.27s
```

**Comment by dicej** [comment]
Now we're seeing Maven being flaky on other platforms: https://github.com/bytecodealliance/wit-bindgen/actions/runs/4197128426/jobs/7279042855#step:13:2556
I'll go ahead and pursue the "fat" jar approach when I have time.

**Comment by dicej** [comment]
My plan to address this is stop using Maven and instead download a couple of TeaVM jar files from https://repo.maven.apache.org and use them directly.

@alexcrichton Would it make more sense to download those jar files lazily in the Rust code or in the GitHub workflow?  If we only do the latter, devs will need to download them manually when running the tests locally.  I'm open to other ideas as well.

**Comment by alexcrichton** [comment]
Perhaps a script which is run on github but can also run locally? That's how I ended up handling downloading Wasmtime for the embeddings like wasmtime-{py,go}

**Comment by dicej** [comment]
This should be fixed as of #520
